frappe/templates/pages/integrations/braintree_checkout.py

```diff
@@ -32,10 +32,23 @@ def get_context(context):
 		frappe.local.flags.redirect_location = frappe.local.response.location
 		raise frappe.Redirect
 
+def validate_payment(reference_doctype, reference_docname):
+	status = frappe.db.get_value(reference_doctype, reference_docname, 'status')
+	if status == 'Paid':
+		frappe.log_error("The Payment Request {0} is already paid, cannot process payment twice".format(reference_docname))
+		return{
+			"redirect_to": frappe.redirect_to_message(_('Server Error'), _("The Payment Request {0} is already paid, cannot process payment twice".format(reference_docname))),
+			"status": 401
+		}
+	else:
+		return
+
 @frappe.whitelist(allow_guest=True)
 def make_payment(payload_nonce, data, reference_doctype, reference_docname):
-	data = json.loads(data)
+	if reference_doctype == 'Payment Request':
```

Comment:
This seems to be ERPNext specific code. The best place to add this would be in ERPNext only, by overriding this API there. Another way would be to make this check configurable here and implement it in ERPNext.

This feels similar to how they were trying to fix the SLA problems via https://github.com/frappe/frappe/pull/13443#pullrequestreview-678133106